### PR TITLE
[RFC] Extend v2 Dial/NewConn to declare IFTypes

### PR DESCRIPTION
**Pull request by krsna1729**
Fixes: #68

If we pre-declare in the connection what interface types are implemented
then during AddSession we can iterate through them and add to
teidSessionMap. This will also solve the ambiguous AddTEID on Session
and Conn as we remove the method defined on Conn

Signed-off-by: Saikrishna Edupuganti <saikrishna.edupuganti@intel.com>

### CODE REVIEW

**Comment by krsna1729** [comment]
TODO: Update the api calls in examples/gw-tester.

**Comment by krsna1729** [comment]
This PR is for discussion

File: v2/conn.go [hunk 1]
```diff
@@ -52,10 +53,11 @@ type Conn struct {
 }
 
 // NewConn creates a new Conn used for server. On client side, use Dial instead.
-func NewConn(laddr net.Addr, counter uint8) *Conn {
+func NewConn(laddr net.Addr, counter uint8, ifTypes []uint8) *Conn {
```
Comment (wmnsk) [inline v2/conn.go]: I think the type of the last argument should be `...uint8` to make it simpler for callers.

Comment (krsna1729) [reply v2/conn.go]: Done

**Comment by krsna1729** [comment]
we can close this since #81 solves the issue.

**Pull request closed without merging**